lib/rules/lightness-notation/index.mjs

```diff
@@ -0,0 +1,163 @@
+import valueParser from 'postcss-value-parser';
+
+import declarationValueIndex from '../../utils/declarationValueIndex.mjs';
+import getDeclarationValue from '../../utils/getDeclarationValue.mjs';
+import isStandardSyntaxValue from '../../utils/isStandardSyntaxValue.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import setDeclarationValue from '../../utils/setDeclarationValue.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'lightness-notation';
+
+const messages = ruleMessages(ruleName, {
+	expected: (unfixed, fixed) => `Expected "${unfixed}" to be "${fixed}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/lightness-notation',
+	fixable: true,
+};
+
+const LIGHTNESS_ZERO_TO_ONE_RANGE_FUNCS = new Set(['oklch', 'oklab']);
+const LIGHTNESS_ZERO_TO_HUNDRED_RANGE_FUNCS = new Set(['lch', 'lab']);
+const LIGHTNESS_FUNCS = new Set([
+	...LIGHTNESS_ZERO_TO_ONE_RANGE_FUNCS,
+	...LIGHTNESS_ZERO_TO_HUNDRED_RANGE_FUNCS,
+]);
+const HAS_LIGHTNESS_COLOR_FUNC = new RegExp(`\\b(?:${[...LIGHTNESS_FUNCS].join('|')})\\(`, 'i');
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary, _secondaryOptions, context) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, {
+			actual: primary,
+			possible: ['percentage', 'number'],
+		});
+
+		if (!validOptions) return;
+
+		root.walkDecls((decl) => {
+			if (!HAS_LIGHTNESS_COLOR_FUNC.test(decl.value)) return;
+
+			let needsFix = false;
+			const parsedValue = valueParser(getDeclarationValue(decl));
+
+			parsedValue.walk((node) => {
+				if (node.type !== 'function') return;
+
+				const functionName = node.value.toLowerCase();
+
+				if (!LIGHTNESS_FUNCS.has(functionName)) return;
+
+				const lightness = findLightness(node);
+
+				if (!lightness) return;
+
+				const { value } = lightness;
+
+				if (!isStandardSyntaxValue(value)) return;
+
+				const dimension = valueParser.unit(value);
+
+				if (!dimension) return;
+
+				const isPercentage = dimension.unit === `%`;
+				const isNumber = dimension.unit === '';
+
+				if (!isPercentage && !isNumber) return;
+
+				if (primary === 'percentage' && isPercentage) return;
+
+				if (primary === 'number' && isNumber) return;
+
+				const fixed =
+					primary === 'percentage'
+						? asPercentage(value, functionName)
+						: asNumber(value, functionName);
+				const unfixed = value;
```

Comment:
minor

You could instead rename it at l57.
i.e. `const { value: unfixed } = lightness;`